pkgs/applications/networking/mailreaders/bluemail/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv
+, lib
+, fetchurl
+, dpkg
+, autoPatchelfHook
+, pango
+, gtk3
+, alsa-lib
+, nss
+, libXdamage
+, libdrm
+, mesa
+, libxshmfence }:
+
+stdenv.mkDerivation rec {
+  pname = "bluemail";
+  version = "1.1.119";
+
+  src = fetchurl {
+    url = "https://download.bluemail.me/BlueMail/deb/BlueMail.deb";
```

Comment:
This URL doesn't include the version. You may have to switch to an archived download if this is the only download link.

Reply:
Not sure how to fix this. Should I mirror this file to Internet Archive for a persistent link?

Reply:
The snap version is usually more up-to-date and has a static link with a revision number.
You can get relevant information for the newest version with the following command:
`curl -H 'X-Ubuntu-Series: 16' https://api.snapcraft.io/api/v1/snaps/details/bluemail | jq '.version,.anon_download_url,.revision' -r`
The download link will usually look something like this:
`https://api.snapcraft.io/api/v1/snaps/download/ZVlj0qw0GOFd5JgTfL8kk2Y5eIG1IpiH_${revision}.snap`
This whole procedure has been done for the Spotify nix package as well, so maybe it can serve as a good reference.

I searched hard and couldn't find solutions for the deb and rpm packages.

Reply:
Thanks @SuperSandro2000 that worked. I would like to prefer this approach since it is already working. I added a comment on how to manually update it in case there's a new version